src/shapes.cpp

```diff
@@ -285,18 +293,24 @@ void Shape::padd(double padding)
 void Sphere::scaleAndPadd(double scale, double padding)
 {
   radius = radius * scale + padding;
+  if (radius < 0)
```

Comment:
I think the check for valid dimensions should be made before modifying the object.